pkg/build/builder/source.go

```diff
@@ -229,6 +225,12 @@ func extractGitSource(gitClient GitClient, gitSource *api.GitBuildSource, revisi
 		if err := gitClient.SubmoduleUpdate(dir, true, true); err != nil {
 			return true, err
 		}
+
+	} else {
+		cloneOpts.Shallow = true
+		if err := gitClient.CloneWithOptions(dir, gitSource.URI, cloneOpts); err != nil {
+			return true, err
+		}
 	}
```

Comment:
I don't understand a bit: before we have always executed `gitClient.CloneWithOptions()` and now we do it only in case of not using ref. Why we don't need to clone when `usingRef` is `true`?